src/app/settings/secrets/EditSecretDialog.tsx

```diff
@@ -0,0 +1,222 @@
+				</DialogHeader>
+				<EditSecret secretId={secretId} onClose={onClose} />
+			</DialogContent>
+		</Dialog>
+	);
+}
+
+interface EditSecretProps {
+	secretId: string;
+	onClose: () => void;
+}
+
+interface KeyValue {
+	key: string;
+	value: string;
+}
+
+export function EditSecret({ secretId, onClose }: EditSecretProps) {
+	const { data: secretDetail, isLoading, isError } = useGetSecretDetail(secretId);
+	const [secretName, setSecretName] = useState<string>("");
+	const [keysValues, setKeysValues] = useState<KeyValue[]>([{ key: "", value: "" }]);
+	const [visibleIndex, setVisibleIndex] = useState<number | null>(null);
+
+	useEffect(() => {
+		if (secretDetail) {
+			setSecretName(secretDetail.name);
+			setKeysValues(
+				Object.entries(secretDetail.body.values || {}).map(([key, value]) => ({
+					key,
+					value: String(value)
+				}))
+			);
+		}
+	}, [secretDetail]);
+
+	const addKeyValuePair = () => {
+		const allFilled = keysValues.every((pair) => pair.key !== "" && pair.value !== "");
+		if (allFilled) {
+			setKeysValues([...keysValues, { key: "", value: "" }]);
+		} else {
+			alert("Please fill out all previous key-value pairs before adding a new one.");
+		}
+	};
+
+	const { toast } = useToast();
+	const queryClient = useQueryClient();
+	const { mutate } = useUpdateSecret({
+		onError(error) {
+			if (isFetchError(error)) {
+				toast({
+					status: "error",
+					emphasis: "subtle",
+					description: error.message,
+					rounded: true
+				});
+			}
+		},
+		onSuccess() {
+			queryClient.invalidateQueries({ queryKey: ["secrets"] });
+			onClose(); // Close dialog on success
+		}
+	});
+
+	const handleInputChange = (index: number, event: React.ChangeEvent<HTMLInputElement>) => {
+		const { name, value } = event.target;
+		const updatedKeysValues = keysValues.map((pair, idx) =>
+			idx === index ? { ...pair, [name]: value } : pair
+		);
+		setKeysValues(updatedKeysValues);
+	};
+
+	const removeKeyValuePair = (index: number) => {
+		if (keysValues.length > 1) {
+			const updatedKeysValues = [...keysValues];
+			updatedKeysValues.splice(index, 1);
+			setKeysValues(updatedKeysValues);
+		}
+	};
+
+	const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+		event.preventDefault();
+		const updatedSecretData: UpdateSecret = {
+			name: secretName,
+			scope: "workspace", // Ensure this is a valid value from components["schemas"]["SecretScope"]
+			values: keysValues.reduce(
+				(acc, pair) => {
+					if (pair.key && pair.value) acc[pair.key] = pair.value;
+					return acc;
+				},
+				{} as Record<string, unknown>
+			)
+		};
+		mutate({ id: secretId, body: updatedSecretData });
+	};
+
+	const togglePasswordVisibility = (index: number) => {
+		setVisibleIndex(visibleIndex === index ? null : index);
+	};
+	if (isLoading) return <p></p>;
+	if (isError) return <p>Error fetching secret details.</p>;
+
+	return (
+		<>
+			<form id="edit-secret-form" className="gap-5 p-5" onSubmit={handleSubmit}>
```

Comment:
Same here, if we use `react-hook-form` with a zod schema, we can get rid of a lot of this custom logic from here

Reply:
Done